src/test/java/it/gov/pagopa/pu/fileshare/service/ingestion/IngestionFlowFileServiceImplTest.java

```diff
@@ -61,22 +75,54 @@ void givenAuthorizedUserWhenUploadIngestionFlowFileThenOk(){
 
     Mockito.when(foldersPathsConfigMock.getIngestionFlowFilePath(IngestionFlowFileType.RECEIPT))
       .thenReturn(receiptFilePath);
-    Mockito.when(fileStorerServiceMock.saveToSharedFolder(organizationId, file,receiptFilePath))
+    Mockito.when(fileStorerServiceMock.saveToSharedFolder(organizationId, file, receiptFilePath))
       .thenReturn(filePath);
     Mockito.when(ingestionFlowFileDTOMapperMock.mapToIngestionFlowFileDTO(file,
-        IngestionFlowFileType.RECEIPT, FileOrigin.PAGOPA, organizationId,filePath))
-        .thenReturn(ingestionFlowFileRequestDTO);
-    Mockito.when(ingestionFlowFileClientMock.createIngestionFlowFile(ingestionFlowFileRequestDTO,accessToken))
+        IngestionFlowFileType.RECEIPT, FileOrigin.PAGOPA, organizationId, filePath))
+      .thenReturn(ingestionFlowFileRequestDTO);
+    Mockito.when(ingestionFlowFileClientMock.createIngestionFlowFile(ingestionFlowFileRequestDTO, accessToken))
       .thenReturn(expectedIngestionFlowFileId);
 
     String result = ingestionFlowFileService.uploadIngestionFlowFile(organizationId, IngestionFlowFileType.RECEIPT, FileOrigin.PAGOPA,
       file, TestUtils.getSampleUser(), accessToken);
 
     Assertions.assertSame(expectedIngestionFlowFileId, result);
-    Mockito.verify(userAuthorizationServiceMock).checkUserAuthorization(organizationId,TestUtils.getSampleUser(),accessToken);
-    Mockito.verify(fileServiceMock).validateFile(file,VALID_FILE_EXTENSION);
-    Mockito.verifyNoMoreInteractions(userAuthorizationServiceMock,fileServiceMock,
-      foldersPathsConfigMock,fileStorerServiceMock,ingestionFlowFileDTOMapperMock,ingestionFlowFileClientMock);
+    Mockito.verify(userAuthorizationServiceMock).checkUserAuthorization(organizationId, TestUtils.getSampleUser(), accessToken);
+    Mockito.verify(fileServiceMock).validateFile(file, VALID_FILE_EXTENSION);
+    Mockito.verifyNoMoreInteractions(userAuthorizationServiceMock, fileServiceMock,
+      foldersPathsConfigMock, fileStorerServiceMock, ingestionFlowFileDTOMapperMock, ingestionFlowFileClientMock);
+  }
+
+  @Test
+  void givenAuthorizedUserWhenDownloadIngestionFlowFileThenReturnFileResource() {
+    String accessToken = "TOKEN";
+    Long organizationId = 1L;
+    Long ingestionFlowFileId = 10L;
+    String sharedFolderPath = "/shared";
+    String filePathName = "examplePath";
+    String fileName = "testFile.zip";
+    String fullFilePath = sharedFolderPath + "/" + organizationId + "/" + filePathName + "/" + ARCHIVED_SUB_FOLDER + "/" + fileName;
+    UserInfo user = TestUtils.getSampleUser();
+
+    IngestionFlowFile ingestionFlowFile = new IngestionFlowFile();
+    ingestionFlowFile.setFileName(fileName);
+    ingestionFlowFile.setFilePathName(filePathName);
+    ingestionFlowFile.setStatus(IngestionFlowFile.StatusEnum.COMPLETED);
+
+    InputStreamResource decryptedResource = new InputStreamResource(new ByteArrayInputStream("file content".getBytes()));
```

Comment:
an InputStream should be closed after the usage, is it necessary to create it or could you use Mockito instead to create it?
```suggestion
    InputStreamResource decryptedResource = Mockito.mock(InputStreamResource.class);
```